src/data/lex.rs

```diff
@@ -445,4 +462,12 @@ pub(crate) mod test {
             assert_eq!(&first.to_string(), *token);
         }
     }
+
+    #[test]
+    fn str_display_escape() {
+        let token = "\"Hello, world\\n\\r\\t\"";
+        let mut lexer = cpp(token);
+        let first = lexer.next().unwrap().unwrap().data;
+        assert_eq!(&first.to_string(), token);
```

Comment:
I forget exactly where to import this from (maybe `crate::parse::test`?), but this is the same thing more concisely. It also makes sure that only one token was lexed, not multiple.
```suggestion
        assert_no_change(token);
```

Reply:
There are 3 `assert_no_change` methods, but they will all try to parse the input as source code. So I guess I will stick to `assert_eq`.

Reply:
I'll probably a version of this for literals in a later PR, but I agree for now `assert_eq` is fine.